gensim/models/word2vec.py

```diff
@@ -1120,7 +1135,8 @@ def add_word(word, weights):
                     weights = fromstring(fin.read(binary_len), dtype=REAL)
                     add_word(word, weights)
             else:
-                for line_no, line in enumerate(fin):
+                for line_no in xrange(vocab_size):
```

Comment:
Without it, any user-specified `limit` cap on `vocab_size` won't be respected for `binary=False`-format reads. 

(Without a `limit`, the only way this would prevent a line from being read is if the top-of-file declared count is smaller than the following number of lines... and in such case, reading exactly the declared number of vectors is still arguably a reasonable thing to do, and would also match the behavior of the `binary=True` branch.)